lib/node_modules/@stdlib/number/float32/base/normalize/test/test.native.js

```diff
@@ -0,0 +1,125 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var resolve = require( 'path' ).resolve;
+var tape = require( 'tape' );
+var pow = require( '@stdlib/math/base/special/pow' );
+var uniform = require( '@stdlib/random/base/uniform' );
+var discreteUniform = require( '@stdlib/random/base/discrete-uniform' );
+var PINF = require( '@stdlib/constants/float64/pinf' );
+var NINF = require( '@stdlib/constants/float64/ninf' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var FLOAT32_SMALLEST_NORMAL = require( '@stdlib/constants/float32/smallest-normal' );
+var FLOAT32_SMALLEST_SUBNORMAL = require( '@stdlib/constants/float32/smallest-subnormal' ); // eslint-disable-line id-length
+var tryRequire = require( '@stdlib/utils/try-require' );
+
+
+// VARIABLES //
+
+var normalize = tryRequire( resolve( __dirname, './../lib/native.js' ) );
+var opts = {
+	'skip': ( normalize instanceof Error )
+};
+
+
+// TESTS //
+
+tape( 'main export is a function', opts, function test( t ) {
+	t.ok( true, __filename );
+	t.equal( typeof normalize, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function normalizes a denormalized number, returning a normal number and an exponent', opts, function test( t ) {
+	var frac;
+	var exp;
+	var x1;
+	var x;
+	var v;
+	var i;
+
+	// Smallest denormalized number:
+	v = normalize( FLOAT32_SMALLEST_SUBNORMAL );
+	t.ok( v[ 0 ] >= FLOAT32_SMALLEST_NORMAL, 'returns a normal number' );
+	t.equal( v[ 0 ]*pow( 2, v[ 1 ] ), FLOAT32_SMALLEST_SUBNORMAL, 'x = y * 2^exp' );
+
+	// Other subnormals...
+	for ( i = 0; i < 1000; i++ ) {
+		frac = uniform( 0.26, 10.0 ); // 0.26 prevents underflow
+		exp = discreteUniform( -323, -309 );
```

Comment:
This is incorrect. See the tests for this package. These values are only appropriate for the double-precision variant.